WordPress/Classes/ViewRelated/Reader/Subscriptions/ReaderSubscriptionsView.swift

```diff
@@ -128,9 +129,66 @@ struct ReaderSubscriptionsView: View {
         ReaderSubscriptionHelper().unfollow(site)
     }
 
-    private func reloadSearchResults(searchText: String) {
-        let ranking = StringRankedSearch(searchTerm: searchText)
-        searchResults = ranking.search(in: subscriptions) { "\($0.title) \($0.siteURL)" }
+    private func performBackgroundSearch(searchText: String) {
+        struct SearchableSubscription: Sendable {
+            let objectID: NSManagedObjectID
+            let title: String
+            let siteURL: String
+
+            var searchableText: String {
+                "\(title) \(siteURL)"
+            }
+
+            init(_ subscription: ReaderSiteTopic) {
+                self.objectID = subscription.objectID
+                self.title = subscription.title
+                self.siteURL = subscription.siteURL
+            }
+        }
+
+        // Cancel any existing search task
+        searchTask?.cancel()
+
+        // Clear results immediately if search text is empty
+        if searchText.isEmpty {
+            searchResults = nil
+            pendingSearchText = nil
+            return
+        }
+
+        // Start new background search task
+        searchTask = Task {
+            // Store the search text we're processing
+            let currentSearchText = searchText
+
+            // Create searchable data on main thread to avoid Core Data context issues
+            let searchableData = subscriptions.map(SearchableSubscription.init)
+
+            // Perform the search on a background queue with parallel processing
+            let resultObjectIDs = await StringRankedSearch(searchTerm: currentSearchText)
```

Comment:
Maybe check cancellation before the `await` call, just in case `performBackgroundSearch` gets called too frequently?